docs/credhub-security-faq.md

```diff
@@ -52,12 +52,12 @@ TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
 
 If CredHub's third-party integration with a Luna SafeNet HSM is
 enabled, then CredHub has no knowledge of key values. If CredHub is
-started in internal encryption mode, CredHub deterministically
-generates a site-specific AES256 key on startup. CredHub concatenates
-a user-defined password from its configuration file with a
+started in internal encryption mode, CredHub recalculates
```

Comment:
Maybe "calculates" is better? It could be the first time it calculates it, if the user just provided a new encryption password. But it will **re**calculate it every time after that.

How can we phrase this?

Reply:
I'm having trouble understanding what key it's talking about here. I was assuming it's the key used to encrypt data at rest, but I see that it's changed on startup, which doesn't fit that idea. Is this key used to communicate with clients?

Reply:
Perfect, thanks for sharing that. That is pretty much exactly the confusion I'm trying to solve. Apparently my first attempt is a failure.

It *is* the key used to encrypt data at rest.

> I see that it's changed on startup

Which part makes you say that? Because it's not.

Reply:
The previous version of the docs says that it "generates a site-specific AES256 key on startup" which IMO is very confusing, as it appears that it creates a new key. But it doesn't. It's the same key every time that is computed. That's how it works.

Reply:
Ah, ok. So saying "recalculates" I think is just as confusing as "generates". I don't understand what you're distinguishing between the first startup and subsequent startups, if the action is the same and the resulting key is the same each time.

Reply:
> the action is the same and the resulting key is the same each time

Yes

> I don't understand what you're distinguishing between the first startup and subsequent startups

Fair enough. So would "calculates" be better then?

Reply:
What made me think it was changed at every startup was "recalculates" and the "randomly generated" part of the salt (as a naive reader). Now that I understand better, I agree that "generate" isn't a good way to explain it. So, yes, I like "calculates". And in the spirit of the "deterministically" part that you removed, maybe a brief explanation like "The key will be the same every time CredHub starts as long as the user-provided password has not changed."